### ISSUE TITLE
Corrupt Excel file when using an ampersand in a table header

### ISSUE THREAD

**Issue by holleymcfly** [dhatim/fastexcel#445]
Hi there,

using an ampersand (&) in the header of a formatted excel table leads to a corrupt file: an error message appears when opening the file, and the table format has been removed.
Setting an ampersand in the data row of a table is no problem.

I've created a test project to demonstrate: [demo](https://github.com/holleymcfly/fastexcel-table).

**Comment by meiMingle** [comment]
Thanks for your feedback, I think I have fixed this issue, but the release date is not guaranteed

**Comment by meiMingle** [comment]
A temporary solution is to escape special characters beforehand.
```java
...
worksheet.range(0, 0, row - 1, getDataRow1().size() - 1).createTable(escape(toArray(getHeadersCorrupt())))
                    .styleInfo().setStyleName("TableStyleMedium3");
...
 ```
```java
...
    private String[] escape(String[] stringArray) {
        for (int i = 0; i < stringArray.length; i++) {
            stringArray[i] = XmlEscapeHelper.escape(stringArray[i]);
        }
        return stringArray;
    }
...
```

**Comment by holleymcfly** [comment]
Hi Tom,

thanks for your investigation and reply.

Unfortunately, your suggested solution doesn't work. I had already tried to replace the ampersand with the escaped characters &amp;. If I do this, the error message still comes up when the excel file is opened, and the header doesn't show the ampersand, but the escaped sequence.

Also, using your escape method (which seems to do the same for the ampersand), doesn't help. (I had to modify your solution slightly to "new XmlEscapeHelper().escape(.)", as the method is not static.

Best,
Holger

**Comment by meiMingle** [comment]
I'm not sure if you've applied the correct escapes because I can't see your code. In my test, the `&` is translated to `&amp;`,and stored in the table1.xml file. You can change the suffix of the generated xlsx file to zip, then unzip it and check the xl/tables/table1.xml file. Check the name property on table>tableColumns>tableColumn to see if the `&` is correctly escaped.

**Comment by holleymcfly** [comment]
Yes, the `&` is translated to `&amp;` , and I can see it in the table1.xml file (I always investigate it with 7zip). But that doesn't make the error disappear.
Have you tried your solution with my sample project I created?

**Comment by meiMingle** [comment]
I just checked the code, and in the latest released version 0.18.0, the XmlEscapeHelper().escape(.) method is indeed not static. I have submitted the code to make changes to this a few months ago. This change is already in the latest code of the master branch, but when it will be released depends on the project owner.

**Comment by meiMingle** [comment]
@ochedru Good to know about this.I'm traveling for work at the moment and don't have much spare time, but I'll check this project out once I get some free time. I like this project and I'll try to contribute to it.
